tests/test_functions.py

```diff
@@ -53,6 +62,10 @@ def test_functions():
     assert dill.loads(dumped_func_d)(1, 2, d2=3) == 6
 
     if is_py3():
+        empty_cell = make_empty_cell()
+        cell_copy = dill.loads(dill.dumps(empty_cell))
+        assert 'empty' in str(cell_copy)
```

Comment:
this is somewhat of a weak test, but I guess acceptable.  Is there a more specific test?

Reply:
I honestly cannot think of any useful functions that return inner functions returning uninitialized variables. I just added the case for the sake of completeness. Perhaps it comes up in more complex cases where another inner function assigns to the variable.

Reply:
It's OK.